test/integration/simulacron.js

```diff
@@ -0,0 +1,403 @@
+          done();
+        });
+      });
+    });
+    if (initClient) {
+      var baseOptions = { contactPoints: ['127.0.0.101'] };
+      client = new Client(utils.extend({}, options.clientOptions, baseOptions));
+      before(client.connect.bind(client));
+      after(client.shutdown.bind(client));
+    }
+    afterEach(simulacronCluster.clear.bind(simulacronCluster));
+    after(simulacronHelper.stop.bind(simulacronHelper));
+
+    var setupInfo = { cluster: simulacronCluster, client: client };
+    return setupInfo;
+  },
+  baseOptions: (function () {
+    return {
+      //required
+      cassandraVersion: helper.getCassandraVersion(),
+      dseVersion: '',
+      clusterName: 'testCluster',
+      activityLog: true,
+      numTokens: 1
+    };
+  })(),
+  baseAddress: 'localhost',
+  defaultPort: 8187,
+  SimulacronCluster: SimulacronCluster
+};
+
+function makeRequest(options, callback) {
+  var request = http.request(options, function(response) {
+    // Continuously update stream with data
+    var body = '';
+    response.on('data', function(d) {
+      body += d;
+    });
+    response.on('end', function() {
+      if (body === '') {
+        callback(null);
+      } else {
+        callback(JSON.parse(body));
+      }
+    });
+  });
+  request.on('error', function(err) {
+    helper.trace(err.message);
+    throw new Error(err);
+  });
+  return request;
+}
+
+function SimulacronCluster() {
+  this.baseAddress = simulacronHelper.baseAddress;
+  this.port = simulacronHelper.defaultPort;
+}
+
+SimulacronCluster.prototype.start = function(dcs, clientOptions, callback) {
+  var self = this;
+  var createClusterPath = '/cluster?data_centers=%s&cassandra_version=%s&dse_version=%s&name=%s&activity_log=%s&num_tokens=%d';
+
+  var options = utils.extend({}, simulacronHelper.baseOptions, clientOptions);
+
+  var urlPath = encodeURI(util.format(createClusterPath, dcs, options.cassandraVersion, options.dseVersion,
+    options.clusterName, options.activityLog, options.numTokens));
+
+  var requestOptions = {
+    host: self.baseAddress,
+    port: self.port,
+    path: urlPath,
+    method: 'POST'
+  };
+  makeRequest(requestOptions, function(data) {
+    self.name = data.name;
+    self.id = data.id;
+    self.dcs = data.data_centers;
+    callback(null);
+  }).end();
+};
+
+SimulacronCluster.prototype.destroy = function(callback) {
+  var self = this;
+  var destroyClusterPath = '/cluster/%d';
+  var options = {
+    host: self.baseAddress,
+    path: encodeURI(util.format(destroyClusterPath, self.id)),
+    port: self.port,
+    method: 'DELETE'
+  };
+  makeRequest(options, function(data) {
+    callback();
+  }).end();
+};
+
+SimulacronCluster.prototype._getPath = function (endpoint, id) {
+  var path = '/' + endpoint + '/' + this.id;
+  if (id) {
+    path += '/' + id;
+  }
+  return encodeURI(path);
+};
+
+SimulacronCluster.prototype.log = function(id, callback) {
+  if (typeof id === 'function') {
```

Comment:
I will find a better pattern for this as is done in a lot of methods (if only callback is passed, assume scope global).   Also there's some other repetition i'd like to clean up.